### PR TITLE
Add space toggles for notifications

### PR DESCRIPTION
**Pull request by mattcasey**
### <samp>🤖 Generated by Copilot at 0d1c8e7</samp>

Refactored the code related to opening the settings dialog from different components and contexts, using a common hook and component. Renamed and simplified some functions and types for consistency and readability. Added a feature to import Discord roles from the settings dialog. Added tests and constants for the notification rules logic.

### WHY
New settings:
image

I also reorganized notifications so it is smaller file sizes

### CODE REVIEW

File: components/settings/SettingsContent.tsx
Comment (mattcasey) [inline components/settings/SettingsContent.tsx:1]: This file was extracted from SettingsDialog.tsx

File: components/settings/space/components/NotificationRules.tsx [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    types: [
+      [
+        { label: 'Bounty suggested (Admins only)' },
+        { label: 'Application submitted (Reviewers only)' },
+        { label: 'Application accepted (Applicants only)' },
+        { label: 'Application rejected (Applicants only)' }
+      ],
+      [
+        { label: 'Work submitted (Reviewers only)' },
+        { label: 'Submission approved (Applicants only)' },
```
Comment (Devorein) [inline components/settings/space/components/NotificationRules.tsx:28]: We have one more notificaiton payment_pending sent to all reviewers except for the one that approved the submission

File: components/settings/space/components/NotificationRules.tsx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+import type { NotificationGroup } from 'lib/notifications/interfaces';
+import type { NotificationRuleOption } from 'lib/notifications/notificationRules';
+
+import type { FormValues } from '../SpaceSettings';
+
+type NotificationType = { label: string; type?: NotificationRuleOption };
+
+type ConfigurableGroups = Extract<NotificationGroup, 'rewards' | 'proposals' | 'polls'>;
+
+const notifications: Record<ConfigurableGroups, { title: string; types: NotificationType[][] }> = {
```
Comment (Devorein) [inline components/settings/space/components/NotificationRules.tsx:16]: We still have card (person assigned) and forum post notifications left

File: lib/notifications/constants.ts [hunk 3]
```diff
@@ -1,2 +1 @@
-export const notificationGroups = ['forum', 'proposals', 'rewards', 'polls'] as const;
-export type NotificationGroup = (typeof notificationGroups)[number];
+export const notificationGroups = ['forum', 'documents', 'proposals', 'rewards', 'polls'] as const;
```
Comment (Devorein) [inline lib/notifications/constants.ts:1]: Do we need to add `cards` as we don't track `person_assigned` in the `document` namespace?

Comment (mattcasey) [reply lib/notifications/constants.ts:1]: i'll ask chris, but i figure this is similar to mentions and space admins won't be able to disable

**Pull request merged**